File: src/Ccmc.Core/Services/DeepLinkBuilder.cs
namespace Ccmc.Core.Services;

/// <summary>Builds ccmc://launch deep links. Inverse of <see cref="DeepLinkParser"/>.</summary>
public static class DeepLinkBuilder
{
    public static string Build(string project, bool newSession = false)
    {
        var uri = $"{DeepLinkParser.Scheme}://launch?project={Uri.EscapeDataString(project)}";
        return newSession ? uri + "&new=true" : uri;
    }
}

File: electron/src/core/links/deepLinkBuilder.ts
/** Builds ccmc://launch deep links. Inverse of deepLinkParser. */

import { SCHEME } from './deepLinkParser'

function build(project: string, newSession: boolean = false): string {
  const uri = `${SCHEME}://launch?project=${encodeURIComponent(project)}`
  return newSession ? uri + '&new=true' : uri
}

export const deepLinkBuilder = { build }
